keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -0,0 +1,730 @@
+1. `serviceAccountName`: (optional) the name of a Kubernetes ServiceAccount in the same namespace.  This field is included to support cloud provider identity integration. When supplied, COSI will call the driver to mint new credentials for this ServiceAccount. 
+1. `bucketRequestName`: (required) the name of the `BucketRequest` associated with this access request. From the bucket request, COSI knows the `Bucket` instance and thus the backend bucket and its properties.
+1. `bucketAccessClassName`: (required) name of the `BucketAccessClass` specifying the desired set of policy actions to be set for a user identity or ServiceAccount.
+1. `bucketAccessName`: name of the bound cluster-scoped `BucketAccess` instance. Set by COSI.
+1. `accessGranted`: if true access has been granted to the bucket. If false then access to the bucket has not been granted.
+
+#### BucketAccess
+
+A cluster-scoped administrative custom resource which encapsulates fields from the `BucketAccessRequest` (BAR) and the `BucketAccessClass` (BAC).  The purpose of the `BucketAccess` (BA) is to serve as an access communication path between provisioners and the central COSI controller.  The COSI controller creates a `BucketAccess` instance for a new `BucketAccessRequest`, and there is a 1:1 mapping between `BucketAccess` and `BucketAccessRequest`.
+
+```yaml
+apiVersion: cosi.io/v1alpha1
+kind: BucketAccess
+metadata: 
+  name: [1]
+  labels:
+    cosi.io/provisioner: [2]
+  finalizers:
+  - cosi.io/finalizer [3]
+ spec:
+  provisioner: [4]
+  bucketInstanceName: [5]
+  bucketAccessRequest: [6]
+  serviceAccount: [7]
+  mintedSecretName: [8]
+  policyActionsConfigMapData: [9]
+  principal: [10]
+  parameters: [11]
+ status:
+   accessGranted: [12]
+```
+
+1. `name`: the BA name is generated by COSI in this format: _"<BAR.namespace*>-<BAR.name*>-<uuid>"_. The uuid is unique within a cluster. `*` the first 100 characters of the namespace and the name are used.
+1. `labels`: added COSI.  Key’s value should be the provisioner name. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
+1. `finalizers`: added by COSI to defer `BucketAccess` deletion until the pod that uses this access terminates, and the related `BucketAccessRequest` is deleted.
+1. `provisioner`: (optional) The provisioner field as defined in the `BucketAccessClass`. If empty then there is no driver/sidecar, thus the admin had to create the `Bucket` and `BucketAccess` instances.
+1. `bucketInstanceName`:  name of the `Bucket` instance bound to this BA.
+1. `bucketAccessRequest`: an `objectReference` containing the name, namespace and UID of the associated `BucketAccessRequest`.
+1. `serviceAccount`: an `objectReference` containing the name, namespace and UID of the associated `ServiceAccount`. Empty when the `BucketAccessRequest.mintedSecretName` is specified.
```

Comment:
How is this different from the ServiceAccount in BucketAccessRequest? Do we need it in both places? What happens if they skew for some reason?

Reply:
BA.serviceAccount is copied from BAR.serviceAccount and converted to an ObjectReference. We need both since the sidecar may need the SA and won't have access to the BAR (due to minimal privileges). There is no check for skew.

In later COSI versions we can handle update events on the BAR and BA and deal with changes to SAs.